backend/elo.py

```
@@ -0,0 +1,111 @@
+        in rank order.
+    """
+    resolved: list[str] = []
+    for label in parsed_ranking:
+        model = label_to_model.get(label)
+        if model and model not in resolved:
+            resolved.append(model)
+
+    pairs: list[tuple[str, str]] = []
+    for i, winner in enumerate(resolved):
+        for loser in resolved[i + 1:]:
+            pairs.append((winner, loser))
```

Ruff (0.15.11)

[warning] Use `list.extend` to create a transformed list

(PERF401)